src/coreComponents/mesh/generators/VTKWellGenerator.cpp

```diff
@@ -56,17 +57,25 @@ void VTKWellGenerator::fillPolylineDataStructure( )
     vtkPoints * points = polyData->GetPoints();
     m_polyNodeCoords.resize( points->GetNumberOfPoints(), 3 );
     globalIndex ipoint = 0;
-    for( vtkIdType c = 0; c < points->GetNumberOfPoints(); ++c )
+    for( vtkIdType c = 0; c < points->GetNumberOfPoints(); ++c, ++ipoint )
     {
       real64 point[3];
       points->GetPoint( c, point );
-      LvArray::tensorOps::copy< 3 >( m_polyNodeCoords[ipoint++], point );
+      LvArray::tensorOps::copy< 3 >( m_polyNodeCoords[ipoint], point );
     }
 
+    GEOS_ERROR_IF( polyData->GetLines()->GetNumberOfCells() == 0, GEOS_FMT( "{}: Error! Your VTK file {} doesn't contain any well",
+                                                                            this->getName(), m_filePath ));
+
+    GEOS_LOG_RANK_0_IF( polyData->GetLines()->GetNumberOfCells() > 1, GEOS_FMT( "{}: Warning! Your VTK file {} contains multiple wells. Only the first one will be read",
+                                                                                this->getName(), m_filePath ));
+
     // load edges
     polyData->GetLines()->InitTraversal();
     vtkNew< vtkIdList > idList;
-    polyData->GetLines()->GetNextCell( idList );
+    integer hadRead = polyData->GetLines()->GetNextCell( idList );
+
+    GEOS_ERROR_IF( hadRead == 0, GEOS_FMT( "{}: Error! Your VTK file {} doesn't contain any line", this->getName(), m_filePath ));
```

Comment:
Is this possible give the check at line 67?

Reply:
Shouldn't